src/main/java/ui/components/pickers/PickerMilestone.java

```diff
@@ -128,15 +172,17 @@ public int compareTo(PickerMilestone milestone) {
             return this.isOpen() ? -1 : 1;
         }
 
-        if (this.getDueDate().equals(milestone.getDueDate())) return 0;
+        // milestones ordered according to their titles if due dates are the same
+        if (this.getDueDate().equals(milestone.getDueDate())) {
+            return this.getTitle().compareTo(milestone.getTitle());
+        }
 
         // milestones with due dates are smaller
         if (!this.getDueDate().isPresent()) return 1;
         if (!milestone.getDueDate().isPresent()) return -1;
 
         // milestones with earlier due dates are smaller
-        return this.getDueDate().get()
-                .isBefore(milestone.getDueDate().get()) ? -1 : 1;
+        return this.getDueDate().get().isBefore(milestone.getDueDate().get()) ? -1 : 1;
     }
 
```

Comment:
Sorry, I missed this one. You need a test to cover this as well, for picker milestone in which both has no due date and both has same due date.